src/utils/OpenPepXLLF.cpp

```diff
@@ -569,130 +562,134 @@ class TOPPOpenPepXLLF :
       vector< OPXLDataStructs::XLPrecursor >::const_iterator low_it;
       vector< OPXLDataStructs::XLPrecursor >::const_iterator up_it;
 
-      if (precursor_mass_tolerance_unit_ppm) // ppm
-      {
-        allowed_error = precursor_mass * precursor_mass_tolerance * 1e-6;
-      }
-      else // Dalton
-      {
-        allowed_error = precursor_mass_tolerance;
-      }
-#ifdef _OPENMP
-#pragma omp critical (enumerated_cross_link_masses_access)
-#endif
-      {
-        low_it = lower_bound(enumerated_cross_link_masses.begin(), enumerated_cross_link_masses.end(), precursor_mass - allowed_error, OPXLDataStructs::XLPrecursorComparator());
-        up_it =  upper_bound(enumerated_cross_link_masses.begin(), enumerated_cross_link_masses.end(), precursor_mass + allowed_error, OPXLDataStructs::XLPrecursorComparator());
-      }
+      vector< double > spectrum_precursor_vector;
+      vector< double > allowed_error_vector;
 
-      if (low_it != up_it) // no matching precursor in data
+      progresslogger.startProgress(0, 1, "Start enumerating candidates...");
+      for (int correction_mass : precursor_correction_steps)
       {
-        for (; low_it != up_it; ++low_it)
+        double allowed_error = 0;
+
+        double corrected_precursor_mass = precursor_mass + (static_cast<double>(correction_mass) * Constants::C13C12_MASSDIFF_U);
+
+        if (precursor_mass_tolerance_unit_ppm) // ppm
+        {
+          allowed_error = corrected_precursor_mass * precursor_mass_tolerance * 1e-6;
+        }
+        else // Dalton
         {
-          candidates.push_back(*low_it);
+          allowed_error = precursor_mass_tolerance;
         }
+
+        spectrum_precursor_vector.push_back(corrected_precursor_mass);
+        allowed_error_vector.push_back(allowed_error);
+
+      } // end correction mass loop
+
+      vector< int > precursor_correction_positions;
+      candidates = OPXLHelper::enumerateCrossLinksAndMasses(filtered_peptide_masses, cross_link_mass, cross_link_mass_mono_link, cross_link_residue1, cross_link_residue2, spectrum_precursor_vector, precursor_correction_positions, precursor_mass_tolerance, precursor_mass_tolerance_unit_ppm);
+
+      vector< int > precursor_corrections;
+      for (Size pc = 0; pc < precursor_correction_positions.size(); ++pc)
+      {
+        precursor_corrections.push_back(precursor_correction_steps[precursor_correction_positions[pc]]);
       }
 
+      vector <OPXLDataStructs::ProteinProteinCrossLink> cross_link_candidates = OPXLHelper::buildCandidates(candidates, precursor_corrections, precursor_correction_positions, filtered_peptide_masses, cross_link_residue1, cross_link_residue2, cross_link_mass, cross_link_mass_mono_link, spectrum_precursor_vector, allowed_error_vector, cross_link_name);
+
+      progresslogger.endProgress();
+
+      cout << "Size of enumerated candidates: " << double(cross_link_candidates.size()) * sizeof(OPXLDataStructs::ProteinProteinCrossLink) / 1024.0 / 1024.0 << " mb" << endl;
```

Comment:
what about these 1024 / 1024?

Reply:
this is just a conversion from bytes to mb and I think this makes it easier to read and understand than   / 1048576    but now that you mentioned it, does it make sense to write this out at all? I used this to see how much memory I saved with some changes, but this will not be very useful for users since it is only one chunk of the used memory anyway, and not always the biggest one. I will make this and some other outputs to debug outputs.